### PR TITLE
Enhance check_sequences_exhaust to take account of all sequences

### PR DESCRIPTION
**Pull request by frost242**
Fixes issue #291.

### CODE REVIEW

File: check_pgactivity [hunk 1]
```diff
@@ -7065,9 +7065,10 @@ PARAM_LOOP: foreach my $row (@rs) {
 
 =item B<sequences_exhausted> (7.4+)
 
-Check all sequences assigned to a column (the smallserial, serial and bigserial
-types), and raise an alarm if the column or sequences gets too close to the
-maximum value.
+Check all sequences and raise an alarm if the column or sequences gets too
+close to the maximum value. The maximum value is calculated from the maxvalue
+of the sequence or from the column type when the sequence is assigned to a
```
Comment (rjuju) [inline check_pgactivity:7070]: Should be "owned" rather than "assigned"?

File: check_pgactivity [hunk 2]
```diff
@@ -7203,15 +7206,18 @@ SELECT
     seq.seqmax AS max_value,
     seq.seqincrement AS increment_by
    FROM pg_sequence seq
-   INNER JOIN pg_catalog.pg_depend d ON d.objid=seq.seqrelid
+   LEFT JOIN (SELECT d.objid, nspname, relname, attname, typname FROM
```
Comment (rjuju) [inline check_pgactivity:7209]: nit: the FROM should be on the next line.

File: check_pgactivity [hunk 3]
```diff
@@ -7241,15 +7248,10 @@ SELECT
...
                 $max_value=9223372036854775807;
             }
-            else
-            {
-                # We're not going to try to guess. This is not a serial, trust
-                # the dba/developer
-                delete $sequences{$seq};
-                next;
-            }
+	    # When we can't link the sequence to a column, we true the dba/developer
```
Comment (rjuju) [inline check_pgactivity:7251]: s/true/trust/

File: check_pgactivity [hunk 4]
```diff
@@ -7264,12 +7266,16 @@ SELECT
                 $sequences{$seq}->{ABSVALSEQ}=-$sequences{$seq}->{LASTVALSEQ};
             }
             # The real maximum value is the smallest of both
-            my $real_max_value=$max_val_seq<=$max_value?$max_val_seq:$max_value;
+	    my $real_max_value;
+	    if ( defined $max_value ) {
+                $real_max_value=$max_val_seq<=$max_value?$max_val_seq:$max_value;
```
Comment (rjuju) [inline check_pgactivity:7271]: if you're going to touch that line, maybe we could add space around operator? it's a bit unreadable as-is

**Review by rjuju — CHANGES_REQUESTED** [review]
New query looks good to me, a few stylistic changes comments only.

File: check_pgactivity (hunk 2 above)
Comment (frost242) [reply check_pgactivity:7209]: Fixed in a forthcoming commit

File: check_pgactivity (hunk 4 above)
Comment (frost242) [reply check_pgactivity]: done in 73caa656410f68e02c2dcbc4499b3bd3add77fa2

File: check_pgactivity (hunk 2 above)
Comment (frost242) [reply check_pgactivity]: done in 73caa656410f68e02c2dcbc4499b3bd3add77fa2

File: check_pgactivity (hunk 3 above)
Comment (frost242) [reply check_pgactivity]: done in 73caa656410f68e02c2dcbc4499b3bd3add77fa2

File: check_pgactivity (hunk 1 above)
Comment (frost242) [reply check_pgactivity]: done in 73caa656410f68e02c2dcbc4499b3bd3add77fa2

**Review by rjuju — APPROVED** [review]
LGTM!

**Pull request merged**